heaps/min_heap.py

```diff
@@ -57,18 +68,43 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(log n)
         """
-        pass
+        if index == 0:
+            return
+
+        if index % 2 == 0:
+            # right node parent
+            parent_index = int((index - 2) / 2)
+        else:
+            # left node parent
+            parent_index = int((index - 1) / 2)
+
+        if(parent_index < 0):
+            return
+
+        if self.store[parent_index].key > self.store[index].key:
+            self.swap(parent_index, index)
+            self.heap_up(parent_index)
 
     def heap_down(self, index):
         """ This helper method takes an index and 
             moves the corresponding element down the heap if it's 
             larger than either of its children and continues until
             the heap property is reestablished.
```

Comment:
This works, but review my comment below to see how you might improve upon this solution ⬇️